src/pages/ContentScripts/features/repo-collection/CollectionModal/CollectionEditor.tsx

```diff
@@ -0,0 +1,338 @@
+import { useRepoCollectionContext } from '../context';
+
+import React, { useEffect, useState } from 'react';
+import {
+  Modal,
+  Col,
+  Row,
+  Button,
+  Form,
+  Input,
+  Table,
+  Divider,
+  Radio,
+} from 'antd';
+import type { ColumnsType } from 'antd/es/table';
+
+interface Values {
+  name: string;
+  quickImport: string;
+}
+
+interface DataType {
+  key: React.Key;
+  name: string;
+  description: string;
+}
+
+interface RepositoryInfo {
+  name: string;
+  description: string;
+}
+
+interface CollectionEditorProps {
+  open: boolean;
+  onCreate: (values: Values, newRepoData: string[] | undefined) => void;
+  onCancel: () => void;
+  isEdit: boolean | undefined;
+  collectionName: string;
+  collectionData: string[];
+}
+
+interface DataSourceType {
+  key: string;
+  name: string;
+  description: string;
+}
+/*
+async function getUserOrOrgRepos(
+  username: string,
+  isOrg: boolean,
+  accessToken: string // GitHub Personal Access Token
+): Promise<RepositoryInfo[]> {
+  try {
+    const apiUrl = isOrg
+      ? `https://api.github.com/orgs/${username}/repos`
+      : `https://api.github.com/users/${username}/repos`;
+
+    const response = await fetch(apiUrl, {
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+      },
+    });
+
+    if (!response.ok) {
+      throw new Error(
+        `GitHub API request failed with status: ${response.status}`
+      );
+    }
+
+    const reposData = await response.json();
+
+    const repositories: RepositoryInfo[] = reposData.map((repo: any) => ({
+      name: repo.name,
+      description: repo.description || '',
+    }));
+
+    return repositories;
+  } catch (error) {
+    console.error('Error fetching repositories:', error);
+    throw error;
+  }
+}
+*/
+// 定义多个 GitHub Personal Access Token
+const accessTokens = [
+  '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',
+  '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',
+  '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',
+  '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',
+  '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',
+]; // 添加你的令牌
```

Comment:
I don't think we should have acessTokens like this. This could cause security problems. At lease move the tokens to the .env file in the root directory. (we currently don't have one, so you might need to create one)

Reply:
Actually we provided an input for users to set their own access token before. This is a preffered way to set secret variables.

And actually these 3 tokens have already been exposed once the commits were pushed into this PR :D

Reply:
> Actually we provided an input for users to set their own access token before. This is a preffered way to set secret variables.
> 
> And actually these 3 tokens have already been exposed once the commits were pushed into this PR :D

We can expire these tokens, so it will be fine. You are right, we should ask users for the tokens, feel like this is the only way. This feature should be delayed for future implementation.